creator-node/src/fileManager.js

```diff
@@ -162,77 +156,73 @@ async function saveFileForMultihashToFS (req, multihash, expectedStoragePath, ga
     // If file not already stored, fetch and store at storagePath.
     let fileFound = false
 
-    // If multihash already available on local ipfs node, cat file from local ipfs node
-    req.logger.debug(`checking if ${multihash} already available on local ipfs node`)
-    try {
-      // ipfsCat returns a Buffer
-      decisionTree.push({ stage: 'About to retrieve file from local ipfs node with cat', vals: multihash, time: Date.now() })
-      let fileBuffer = await Utils.ipfsCat(multihash, req, 1000)
-      fileFound = true
-      req.logger.debug(`Retrieved file for ${multihash} from  with cat`)
-      decisionTree.push({ stage: 'Retrieved file from local ipfs node with cat', vals: multihash, time: Date.now() })
-      // Write file to disk.
-      await writeFile(expectedStoragePath, fileBuffer)
-      req.logger.info(`wrote file to ${expectedStoragePath}, obtained via ipfs cat`)
-      decisionTree.push({ stage: 'Wrote file to disk', vals: expectedStoragePath, time: Date.now() })
-    } catch (e) {
-      req.logger.warn(`Multihash ${multihash} is not available on local ipfs node ${e.message}`)
-      decisionTree.push({ stage: 'File not available on local ipfs node with cat', vals: multihash, time: Date.now() })
-    }
-
-    // If file not already available on local ipfs node, fetch from IPFS.
-    if (!fileFound) {
-      req.logger.debug(`Attempting to get ${multihash} from IPFS`)
-      try {
-        decisionTree.push({ stage: 'About to retrieve file from local ipfs node with get', vals: multihash, time: Date.now() })
-        // ipfsGet returns a BufferListStream object which is not a buffer
-        // not compatible into writeFile directly, but it can be streamed to a file
-        let fileBL = await Utils.ipfsGet(multihash, req, 1000)
-        req.logger.debug(`retrieved file for multihash ${multihash} from local ipfs node`)
-        decisionTree.push({ stage: 'Retrieved file from local ipfs node with get', vals: multihash, time: Date.now() })
-
-        // Write file to disk.
-        await Utils.writeStreamToFileSystem(fileBL, expectedStoragePath)
-        fileFound = true
-        req.logger.info(`wrote file to ${expectedStoragePath}, obtained via ipfs get`)
-        decisionTree.push({ stage: 'Wrote file to disk', vals: expectedStoragePath, time: Date.now() })
-      } catch (e) {
-        req.logger.warn(`Failed to retrieve file for multihash ${multihash} from IPFS ${e.message}`)
-        decisionTree.push({ stage: 'File not available on local ipfs node with ipfs get', vals: multihash, time: Date.now() })
-      }
-    }
-
-    // if file is still null, try to fetch from other cnode gateways if user has nodes in replica set
-    if (!fileFound && gatewayUrlsMapped.length > 0) {
+    // First try to fetch from other cnode gateways if user has non-empty replica set.
```

Comment:
this diff is very confusing but contains essentially:
1) re-order gateway retrieval to be before ipfs retrieval
2) change gateway retrieval from sequential (for-loop) to parallel (batched for-loop with `Promise.any()`)

Reply:
prob easier to just checkout code / review without this side-by-side diff